src/components/Comments.jsx

```diff
@@ -1,28 +1,67 @@
 import { useEffect, useState } from "react"
 import axios from "axios";
 import CommentCard from "./CommentCard";
+import { getCommentsByArticleId, postNewComment } from "../api/apiFunctions";
+
 
 function Comments({article_id}) {
 
     const [comments, setComments] = useState([])
+    const [newComment, setNewComment] = useState("")
+    const [loggedInUser] = useState("jessjelly")
+    const [commentData, setCommentData] = useState({})
+    const [commentPending, setCommentPending] = useState(false)
+    const [refresh, setRefresh] = useState(0)
+
+    function handleSubmitComment(e) {
+        setCommentPending(true)
+        e.preventDefault()
+        postNewComment(article_id, commentData)
+        .then((response)=>{
+            setCommentPending(false)
+            setNewComment("")
+            setRefresh(Math.random())
+        })
+        .catch((error)=>{
+            setCommentPending(false)
+        })
+    }
+
+    useEffect(()=>{
+        setCommentData({
+            username: loggedInUser, 
+            body: newComment
+        })
+    }, [newComment])
 
     useEffect(()=>{
```

Comment:
rather than making a new request to add the newly posted comment to the comments array, consider optimistically rendering them to reduce the number of requests made to the API: https://notes.northcoders.com/courses/js-front-end/ui-behaviours-and-optimistic-rendering#optimistic-rendering